hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -132,9 +132,8 @@ private static boolean isSupportedCharacter(char c, boolean isStrictS3) {
     // ozone allows namespace to follow other volume/bucket naming convention,
     // for example, here supports '_',
     // which is a valid character in POSIX-compliant system, like HDFS.
-    return (c == '.' || c == '-' ||
-        Character.isLowerCase(c) || Character.isDigit(c)) ||
-        (c == '_' && !isStrictS3);
+    return (OzoneConsts.SUPPORTED_CHARACTER_CHECK_REGEX
+        .matcher(Character.toString(c)).matches()) || (c == '_' && !isStrictS3);
```

Comment:
Since now we are now running the regex over each character rather than a whole string. Could it be a lot slower?

In theory, we could check the whole name string **once** with the regex (with an extra `+` at the end of course) in `verifyResourceName` to avoid checking **each** character one by one. But that would require some refactoring on the whole resource name checking logic, and need to avoid disrupting the functionality of the `isStrictS3` config.

@Tejaskriya Can we do a quick unit test benchmark of the `isSupportedCharacter` method, e.g. run this method over a 1 million characters loop, and compare with the old checker, to see if it is indeed slower? And if so, by how much.